### ISSUE TITLE
Non-ASCII audio device name causes error

### ISSUE THREAD

**Issue by hsogo** [psychopy/psychopy#1399]
Hi,

I found that PsychoPy raises exception when opening Preference Dialog on Japanese Windows 10.  Error message is following:

> C:\Program Files (x86)\PsychoPy2\lib\site-packages\psychopy\app\preferencesDlg.py:377: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
>   if thisDevName not in options:
> Traceback (most recent call last):
>   File "C:\Program Files (x86)\PsychoPy2\lib\site-packages\psychopy\app\_psychopyApp.py", line 611, in showPrefs
>     prefsDlg = PreferencesDlg(app=self)
>   File "C:\Program Files (x86)\PsychoPy2\lib\site-packages\psychopy\app\preferencesDlg.py", line 159, in \_\_init\_\_
>     specSection=self.prefsSpec[section])
>   File "C:\Program Files (x86)\PsychoPy2\lib\site-packages\psychopy\app\preferencesDlg.py", line 264, in makePrefPage
>     parent=panel, name=pLabel, value=thisPref, spec=thisSpec)
>   File "C:\Program Files (x86)\PsychoPy2\lib\site-packages\psychopy\app\preferencesDlg.py", line 391, in \_\_init\_\_
>     self.valueCtrl = wx.Choice(self.parent, choices=labels)
>   File "C:\Program Files (x86)\PsychoPy2\lib\site-packages\wx-3.0-msw\wx\_controls.py", line 501, in \_\_init\_\_
>     _controls_.Choice_swiginit(self,_controls_.new_Choice(*args, **kwargs))
> UnicodeDecodeError: 'cp932' codec can't decode bytes in position 7-8: illegal multibyte sequence

This occurs due to Non-ASCII characters in audio device names. Inserting following codes to line 377 of preferencesDlg.py, Preference Dialog can be opened and audio device names are saved to userPrefs.cfg in utf-8 encoding.

```
                        devs = sound.getDevices('output')
                        for thisDevName in devs:
                            import sys  ### inserted ###
                            thisDevName = thisDevName.decode(sys.getfilesystemencoding()) ### inserted ###
                            if thisDevName not in options:
                                options.append(thisDevName)
```
Saving Non-ASCII audio device name to userPrefs.cfg causes another problem.  After selecting Non-ASCII audio device in the Preference Dialog, `import psychopy.sound` becomes to raise following exception.

> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "C:\Users\Hiroyuki\Dropbox\WinPython-PsychoPy\python-2.7.10\lib\site-packages\psychopy\sound\\_\_init\_\_.py", line 144, in <module>
>     "{}".format(dev, devNames))
> UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-4: ordinal not in range(128)

This occurs because the audio device name saved in userPrefs.cfg doesn't match the name returned by getDevices() due to encoding inconsistency.

A possible solution is to make getDevices() return device name in Unicode, but I'm not sure whether this is the best.  I'd appreciate it if you could give me some advice.

--
Hiroyuki

**Comment by hsogo** [comment]
I tested pysoundcard, sounddevice and pyo and found that pysoundcard and sounddevice return audio device names in unicode.  So, I think it would be better to modify pyo to return device names in unicode.
This fixes the exception when opening Preference Dialog.

The exception when importing psychopy.sound is another problem.  In short, this exception occurs because logging.error message in line 142-143 of psychopy/sound/\_\_init\_\_.py is str while device names are unicode.
This can be fixed by modifying the message from str to unicode.

These fixes works on my Japanese Windows 10 and Japanece Ubuntu 16.04 LTS.

**Comment by hoechenberger** [comment]
Closed by #1403.
